### PR TITLE
Mock e2e stories

### PR DESCRIPTION
**Pull request by kgroschoff**
### What this PR does / why we need it
Mock e2e tests for
- admin-settings/**administrators**
- admin-settings/**cluster-settings**
- admin-settings/**custom-links**
- admin-settings/**opa-integration**

Also mocked **opa** story, but it is failing due to a problem with machine deployments, so I kept it disabled atm.

### Which issue(s) this PR fixes

Related to #3712

### Release note

```release-note
NONE
```

### CODE REVIEW

File: cypress/plugins/index.ts [hunk 1]
```diff
@@ -23,13 +23,7 @@ export default async (on, config) => {
...
   if (isAPIMocked) {
     // TODO: Remove it after configuring mocks.
-    config.ignoreTestFiles = [
-      '**/integration/stories/opa.spec.ts',
-      '**/integration/stories/admin-settings/administrators.spec.ts',
-      '**/integration/stories/admin-settings/cluster-settings.spec.ts',
-      '**/integration/stories/admin-settings/custom-links.spec.ts',
-      '**/integration/stories/admin-settings/opa-integration.spec.ts',
-    ];
+    config.ignoreTestFiles = ['**/integration/stories/opa.spec.ts'];
```
Comment (kgroschoff) [inline cypress/plugins/index.ts:26]: The machine deployment is disappearing after a short amount of time and therefor the test is failing, see [Image](https://minio.worker.ci.k8c.io/dashboard-cypress-artifacts/pr-3821-pre-dashboard-test-e2e-1456523126740357120/screenshots/stories/opa.spec.ts/OPA%20Story%20--%20should%20have%20healthy%20machine%20deployment%20%28failed%29.png) / [Video](https://minio.worker.ci.k8c.io/dashboard-cypress-artifacts/pr-3821-pre-dashboard-test-e2e-1456523126740357120/videos/stories/opa.spec.ts.mp4)
Not sure why this is happening, if you have any ideas feel free to adjust. In the meantime I just kept the story disabled.

You also then need to add 
```
"opaIntegration": {
  "enabled": true
},
```
to the clusters fixture, to enable opa properly.

**Comment by kgroschoff** [comment]
PTAL @maciaszczykm / @floreks

File: cypress/fixtures/alibaba/constraint.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+      "namespaceSelector": {}
+    },
+    "parameters": {
+      "rawJSON": "{\"labels\":[\"gatekeeper\"]}"
+    },
+    "selector": {
+      "labelSelector": {}
+    }
+  }
+}
```
Comment (maciaszczykm) [inline cypress/fixtures/alibaba/constraint.json:22]: What's the difference between it and Anexia constraints etc.?

File: cypress/integration/stories/opa.spec.ts [hunk 3]
```diff
@@ -126,18 +138,21 @@ describe('OPA Story', () => {
     ClustersPage.getTabCard('OPA Gatekeeper Config').should(Condition.Exist);
   });
 
-  it('should wait for initial machine deployment to be created and healthy', () => {
+  it('should wait for initial machine deployment to be created', () => {
     TrafficMonitor.newTrafficMonitor().method(RequestType.GET).url(Endpoint.MachineDeployments).interceptAndWait();
     ClustersPage.getMachineDeploymentList().should(Condition.Contain, initialMachineDeploymentName);
```
Comment (maciaszczykm) [inline cypress/integration/stories/opa.spec.ts:143]: Traffic monitor might be impacting the mocks. I'm not sure about it but I'd remove it and verify it again. It should not be required as line number 143 should wait for machine deployment.

File: cypress/integration/stories/admin-settings/administrators.spec.ts [hunk 4]
```diff
@@ -34,36 +38,28 @@ describe('Admin Settings - Administrators Story', () => {
...
+    AdminSettings.AdministratorsPage.verifyAdminCount(adminsCount);
   });
 
   it('should add second admin', () => {
-    AdminSettings.AdministratorsPage.getAddAdminBtn().click();
-    AdminSettings.AdministratorsPage.getAddAdminDialogEmailInput()
-      .type(Config.userEmail())
-      .should(Condition.HaveValue, Config.userEmail());
-    AdminSettings.AdministratorsPage.getAddAdminDialogSaveBtn().click();
+    if (Mocks.enabled()) {
```
Comment (maciaszczykm) [inline cypress/integration/stories/admin-settings/administrators.spec.ts:45]: If the code to add admin is already moved to `addAdmin` method please move also the part that is checking if mocks are enabled there too.

**Comment by maciaszczykm** [comment]
Looks pretty good, let's try to make OPA work too. I have added some remarks.

File: cypress/fixtures/alibaba/constraint.json (hunk 2 above)
Comment (kgroschoff) [reply cypress/fixtures/alibaba/constraint.json:22]: actually none, you're right, it is not needed to add it to every single provider. will change

File: cypress/integration/stories/opa.spec.ts (hunk 3 above)
Comment (kgroschoff) [reply cypress/integration/stories/opa.spec.ts:143]: Okay, will try 👍

Comment (maciaszczykm) [reply cypress/integration/stories/opa.spec.ts:140]: If is doesn't work leave the comment and then we can go back to it at some point.

**Comment by maciaszczykm** [comment]
@kgroschoff Rebase it to get the latest stability updates. Is it ready to take a look?

**Comment by kgroschoff** [comment]
@maciaszczykm Generally yes, but opa still has some issues, see https://minio.worker.ci.k8c.io/dashboard-cypress-artifacts/pr-3821-pre-dashboard-test-e2e-ce-1461015350907768832/videos/stories/opa.spec.ts.mp4
For some reason the gatekeeper config isn't appearing in the tab. Seems like the gatekeeper endpoint isn't called again after creation. 🤔 Do you have any idea how to fix it?

File: cypress/integration/stories/opa.spec.ts [hunk 5]
```diff
@@ -184,6 +192,9 @@ describe('OPA Story', () => {
   });
 
   it('should add gatekeeper config', () => {
+    if (Mocks.enabled()) {
+      Mocks.gatekeeperConfig = Mocks.defaultGatekeeperConfig;
```
Comment (maciaszczykm) [inline cypress/integration/stories/opa.spec.ts:196]: Can you try to copy it to the next check too? Perhaps it's overridden somewhere. I'd check that first.

Comment (kgroschoff) [reply cypress/integration/stories/opa.spec.ts:196]: yes, already tried, because I thought the same, but nothing changed 😞

**Comment by kgroschoff** [comment]
closing in favor of https://github.com/kubermatic/dashboard/pull/4139

**Pull request closed without merging**